tests/unit/test_dry_run.py

```
@@ -3,6 +3,98 @@
 import numpy.testing as npt
 import unittest.mock
 
+from .conftest import RINGTEST_DIR
+
+
```

I think dry_run should fail/raise an error when we try to use it with an ngv test. May you add this test?

---

I run a test with dry_run option with the ngv circuit ("tests/simulation/ngv") and it worked. I don't know what was the issue with ngv! Can you or @WeinaJi give me any hint?

---

Oh, I just remember that it was not working with ngv at all. From what I recall from the discussions with Antonio the problem was that the astrocytes were not accounted for. Our ngv test is not testing dry run in any way. Even if we were testing that the circuit is ... broken so I am not sure of the results. To be on the safe side I would:
- throw an error to prevent dry_run with ngv completely
- add an issue to re-enable it once we have time

---

... or did you try the ngv circuit and check if the results are indeed correct?

---

because I suspect dry-run (maybe) works but it returns bad results

---

I added a ticket to address it once we have a unit test ngv circuit

---

May throw an error when trying dry run over ngv circuit is a good idea. What do you think @WeinaJi ?

---

Yes, you could throw an error before loading the cells. I see `self._is_ngv_run` is set at the early stage.
Maybe you can make a dry run test with the ngv config file `tests/simulations/ngv/simulation_config.json` and check `pytest.raises`. If we don't create ngv cells before the exception, it should be ok to run ngv circuit with neocortex models.

---

done